app/src/main/java/com/sschuraytz/charcoaldrawing/MainActivity.java

```diff
@@ -95,4 +99,24 @@ public void onClick(View v) {
             }
         });
     }
+
+    public void setUpUndo()
+    {
+        undoButton = (ImageButton) findViewById(R.id.undoButton);
+        undoButton.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                drawingView.undo();
+            }
+        });
+    }
+
+    public void setUpRedo()
+    {
+        redoButton = (ImageButton) findViewById(R.id.redoButton);
+        redoButton.setOnClickListener(new View.OnClickListener() {
```

Comment:
You should use a lambda. Android Studio should offer to replace this with one.

Reply:
> You should use a lambda. Android Studio should offer to replace this with one.

I upgraded from Language level 7 to 8 to change this. Do you know if that affects anything on the user end?

Reply:
It doesn't affect the user. It is just cleaner code.